packages/ra-ui-materialui/src/input/RadioButtonGroupInput.js

```diff
@@ -78,70 +76,41 @@ const styles = createStyles({
  * The object passed as `options` props is passed to the material-ui <RadioButtonGroup> component
  */
 export const RadioButtonGroupInput = ({
-    classes,
-    className,
-    label,
-    resource,
-    source,
-    input,
-    isRequired,
     choices,
-    options,
-    meta,
     helperText,
+    label,
+    onBlur,
+    onChange,
+    onFocus,
+    options,
     optionText,
     optionValue,
+    resource,
+    source,
     translateChoice,
+    validate,
     ...rest
 }) => {
-    const translate = useTranslate();
+    const classes = useStyles({});
 
-    const handleChange = useCallback(
-        (event, value) => {
-            input.onChange(value);
-        },
-        [input]
-    );
-
-    const renderRadioButton = useCallback(
-        choice => {
-            const choiceName = React.isValidElement(optionText) // eslint-disable-line no-nested-ternary
-                ? React.cloneElement(optionText, { record: choice })
-                : typeof optionText === 'function'
-                ? optionText(choice)
-                : get(choice, optionText);
-
-            const nodeId = `${source}_${get(choice, optionValue)}`;
-
-            return (
-                <FormControlLabel
-                    htmlFor={nodeId}
-                    key={get(choice, optionValue)}
-                    value={get(choice, optionValue)}
-                    control={<Radio id={nodeId} color="primary" />}
-                    label={
-                        translateChoice
-                            ? translate(choiceName, { _: choiceName })
-                            : choiceName
-                    }
-                />
-            );
-        },
-        [optionText, optionValue, source, translate, translateChoice]
-    );
-
-    if (typeof meta === 'undefined') {
-        throw new Error(
-            "The RadioButtonGroupInput component wasn't called within a redux-form <Field>. Did you decorate it and forget to add the addField prop to your component? See https://marmelab.com/react-admin/Inputs.html#writing-your-own-input-component for details."
-        );
-    }
-
-    const { touched, error } = meta;
+    const {
+        id,
+        input,
+        isRequired,
+        meta: { error, touched },
+    } = useInput({
+        onBlur,
+        onChange,
+        onFocus,
+        resource,
+        source,
+        validate,
+        ...rest,
+    });
 
     return (
         <FormControl
             component="fieldset"
-            className={className}
```

Comment:
why did you remove the ability to style the main component? It's an unnecessary breaking change.

Reply:
I didn't remove it, it's in the `rest` prop